cfme/tests/intelligence/reports/test_metering_report.py

```diff
@@ -0,0 +1,314 @@
+            "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+            vm.perf_capture('realtime', 2.hour.ago.utc, Time.now.utc)\""
+            .format(provider.id, repr(vm_name)))
+        assert rc == 0, "Failed to capture VM C&U data:".format(out)
+
+        with appliance.db.client.transaction:
+            result = (
+                appliance.db.client.session.query(metrics.id)
+                .join(ems, metrics.parent_ems_id == ems.id)
+                .filter(metrics.capture_interval_name == 'realtime',
+                metrics.resource_name == vm_name,
+                ems.name == provider.name, metrics.timestamp >= date.today())
+            )
+
+        for record in appliance.db.client.session.query(metrics).filter(
+                metrics.id.in_(result.subquery())):
+            if (record.cpu_usagemhz_rate_average or
+               record.cpu_usage_rate_average or
+               record.derived_memory_used or
+               record.net_usage_rate_average or
+               record.disk_usage_rate_average):
+                return True
+        return False
+
+    wait_for(verify_records_metrics_table, [appliance, provider], timeout=600,
+        fail_condition=False, message='Waiting for VM real-time data')
+
+    # New C&U data may sneak in since 1)C&U server roles are running and 2)collection for clusters
+    # and hosts is on.This would mess up our Chargeback calculations, so we are disabling C&U
+    # collection after data has been fetched for the last hour.
+
+    appliance.server.settings.disable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
+    rc, out = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_rollup_range(4.hour.ago.utc, Time.now.utc,'realtime')\"".
+        format(provider.id, repr(vm_name)))
+    assert rc == 0, "Failed to rollup VM C&U data:".format(out)
+
+    wait_for(verify_records_rollups_table, [appliance, provider], timeout=600, fail_condition=False,
+        message='Waiting for hourly rollups')
+
+    # Since we are collecting C&U data for > 1 hour, there will be multiple hourly records per VM
+    # in the metric_rollups DB table.The values from these hourly records are summed up.
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        if (record.cpu_usagemhz_rate_average or
+           record.cpu_usage_rate_average or
+           record.derived_memory_used or
+           record.net_usage_rate_average or
+           record.disk_usage_rate_average):
```

Comment:
end of the block I'm talking about at L197 that matches `verify_records_rollups_table`